app/api/controllers/validators/resolvers/ResolveNino.scala

```diff
@@ -21,17 +21,23 @@ import api.models.errors.{MtdError, NinoFormatError}
 import cats.data.Validated
 import cats.data.Validated.{Invalid, Valid}
 
-object ResolveNino extends Resolver[String, Nino] {
+object ResolveNino {
+
+  def apply(value: String): Validated[Seq[MtdError], Nino] =
```

Comment:
Why do we bring out the code from the Nino object? Not sure why we can't keep it to this:
```scala
object ResolveNino {

  private val ninoRegex =
    ("^([ACEHJLMOPRSWXY][A-CEGHJ-NPR-TW-Z]|B[A-CEHJ-NPR-TW-Z]|G[ACEGHJ-NPR-TW-Z]|" +
      "[KT][A-CEGHJ-MPR-TW-Z]|N[A-CEGHJL-NPR-SW-Z]|Z[A-CEGHJ-NPR-TW-Y])[0-9]{6}[A-D ]?$").r

  def apply(value: String): Validated[Seq[MtdError], Nino] = {
    if (Nino.isValid(value) && ninoRegex.matches(value)) {
      Valid(Nino(value))
    }
    else {
      Invalid(List(NinoFormatError))
    }
  }

}
```

Reply:
But your wider point about the actual Nino object is a good one. There was a conversation started earlier today about why we do this extra validation in ResolveNino. No resolution yet, but hopefully it will get sorted in BSAS and rolled out if there is any change required.

Reply:
I'm still confused so a call about this might be useful